hardware/plugins/PluginTransports.cpp

```diff
@@ -67,7 +67,7 @@ namespace Plugins {
 				//
 				//	Async resolve/connect based on http://www.boost.org/doc/libs/1_45_0/doc/html/boost_asio/example/http/client/async_client.cpp
 				//
-				m_Resolver.async_resolve(query, [this](const auto &err, auto end) { handleAsyncResolve(err, end); });
+				m_Resolver.async_resolve(query, [this](auto &&err, auto end) { handleAsyncResolve(err, end); });
```

Comment:
&&err ? Are you sure ? (Just double checking)

Reply:
Yeah. It's shorter and implicitly convertable to const auto ref.

Reply:
Actually passing auto&& for a lambda parameter doesn't work right for a function accepting && as a parameter. You still have to call std::move.

Reply:
Is && not the same as a pointer ?

Reply:
It's an rvalue reference. Also known as a universal reference since it implicitly converts to many things.

Note that the clang-tidy modernize-avoid-bind check uses then by default. I didn't originally use them for C++11 compatibility (no auto in lambdas)